lib/matplotlib/tests/test_tightlayout.py

```diff
@@ -288,13 +288,16 @@ def test_badsubplotgrid():
 
 
 def test_collapsed():
-    # test that if a call to tight_layout will collapses the axes that
-    # it does not get applied:
+    # test that if the amount of space required to make all the axes
+    # decorations fit would mean that the actual Axes would end up with size
+    # zero (i.e. margins add up to more than the available width) that a call
+    # to tight_layout will not get applied:
     fig, ax = plt.subplots(tight_layout=True)
     ax.set_xlim([0, 1])
     ax.set_ylim([0, 1])
 
-    ax.annotate('BIG LONG STRING', xy=(1.25, 2), xytext=(10.5, 1.75),)
+    ax.annotate('BIG LONG STRING', xy=(1.25, 2), xytext=(10.5, 1.75),
+                annotation_clip=False)
```

Comment:
This test was actually "buggy" as originally written. Because of the value of `xy`, with `annotation_clip=None`, this text never got drawn, and as such should never have contributed to the `tight_layout` in the first place. The code fixed this, causing this test to fail (since it no longer tested what it was intended to). 

Adding `annotation_clip=False` fixes the test to check for what it originally intended (by actually *drawing* the text way off screen).